cloud-platform/env-create.md

```diff
@@ -0,0 +1,232 @@
+
+Using a Kubernetes namespace could isolate namespaces for different environments in the same cluster. Providing us with flexibility of creating segregated environments for different.
+
+#### Namespace.yaml
+
+```
+apiVersion: 
+kind: 
+metadata:
+      name: This is where you will define your $servicename-$env
+      labels:
+           name: Also define your $servicename-$env
+```
+
+[https://kubernetes.io/docs/tasks/administer-cluster/namespaces-walkthrough/](https://kubernetes.io/docs/tasks/administer-cluster/namespaces-walkthrough/)
+
+#### k8s-non-prod-environment/namespaces/$servicename-$env/$service-$name-admin-role.yaml
+
+```
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: myapp-dev-admins
+  namespace: myapp-dev
+subjects:
+  - kind: Group
+    name: "github:$yourTeam"
+    apiGroup: rbac.authorization.k8s.io
+roleRef:
+  kind: ClusterRole
+  name: admin
+  apiGroup: rbac.authorization.k8s.io
+```
+
+#### Rolebinding:
+
+We will also create a Rolebinding resouce that will provide us with access policies to the namespace we have created in the cluster.
+
+A rolebinding resource grants the permissions defined in a role to a user or set of users. A role can be another resource we can create but in this instance we will reference (roleRef) a Kubernetes default role "ClusterRole - admin" .
+
+This Rolebinding resource references the ClusterRole to provide ClusterRole admin access to a set of users which is defined under subjects. In this case, the `$yourTeam` github group will have admin access to any resources within the namespace "myapp dev".
+
+`$servicename-$env-admin-role.yaml`
+
+Example of yaml above, i will explain parts that need to be changed.
+
+```
+kind: 
+apiVersion:
+metadata:
+     name:
+     namespace: Define your namespace as created in namespace.yaml, this is where its mapped.
+subjects:
+     kind: 
+     name: This is where you specify your Github team, as format shown above "github:$yourTeam"
+     apiGroup:
+roleRef: 
+     kind: 
+     name:
+     apiGroup:
+```
+
+[https://kubernetes.io/docs/admin/authorization/rbac/](https://kubernetes.io/docs/admin/authorization/rbac/)
+
+```
+within our $servicename-$env directory, you will need to create the two files shown above. You can use any file editor of your choice (nano, vim).
+
+$ cd myapp-dev/       								# Make sure you are in your service directory.
+$ vi namespace.yaml   								#create the namespace.yaml and use guide above to help structure file.
+$ vi myapp-dev-admin-role.yaml 						#once completed do the same for admin-role file.
+$ ls                            					#list files in $Servicename-$dev directory, to see both files have been saved.
+myapp-dev-admin-role.yaml	namespace.yaml   		#output after running list command.
+
+##Remember these are Yaml files, so make sure syntax is correct. If syntax is incorrect then file may not build properly, you can use a online validator.
```

Comment:
As an aside - this strongly suggests we should have a YAML linter and/or kube API object validator running on each PR.